tests/Feature/Http/Controllers/EntryControllerTest.php

```diff
@@ -270,6 +271,112 @@ public function store_saves_entry_with_items()
         $this->assertCount(2, $entry->items);
     }
 
+    /**
+     * @test
+     */
+    public function store_handles_dm_reward_choice()
+    {
+        // Set Default Values for All Required Attributes of Entry
+        $adventure = Adventure::factory()->create();
+        $campaign = Campaign::factory()->create();
+        $character_adv = Character::factory()->create(['user_id' => $this->user->id]);
+        $character_MI = Character::factory()->create(['user_id' => $this->user->id]);
+        $character_CR = Character::factory()->create(['user_id' => $this->user->id]);
+        $event = Event::factory()->create();
+        $dungeon_master_user = User::factory()->create();
+        $dungeon_master = $this->faker->word;
+        $date_played = $this->faker->dateTime();
+        $location = $this->faker->word;
+        $type = Entry::TYPE_DM;
+        $levels = $this->faker->numberBetween(1, 20);
+        $gp = $this->faker->randomFloat(2, 0, 9999999.99);
+        $itemData = [
+            ['name' => "Longsword +1", 'rarity' => "uncommon"],
+            ['name' => "Amulet of Health", 'rarity' => "rare", 'description' => "Your Constitution score is 19 while you wear this amulet."]
+        ];
+        $choice_advancement = 'advancement';
+        $choice_item = 'magic_item';
+        $choice_rewards = 'campaign_reward';
+
+        //post request entry.store for each of them
+        $response_advancement = $this->actingAs($this->user)->post(route('entry.store'), [
+            'user_id' => $this->user->id,
+            'adventure_id' => $adventure->id,
+            'campaign_id' => $campaign->id,
+            'character_id' => $character_adv->id,
+            'event_id' => $event->id,
+            'dungeon_master_id' => $dungeon_master_user->id,
+            'dungeon_master' => $dungeon_master,
+            'date_played' => $date_played,
+            'location' => $location,
+            'type' => $type,
+            'levels' => $levels,
+            'gp' => $gp,
+            'choice' => $choice_advancement,
+
```

Comment:
we can probably extract this to its own array since its the same in all three post requests, lets do that in a refactor of this feature next sprint